zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -120,12 +129,31 @@ public PrepRequestProcessor(ZooKeeperServer zks, RequestProcessor nextProcessor)
             + "):", zks.getZooKeeperServerListener());
         this.nextProcessor = nextProcessor;
         this.zks = zks;
+        initMaxEphemeralNodesPerSession();
         this.digestEnabled = ZooKeeperServer.isDigestEnabled();
         if (this.digestEnabled) {
             this.digestCalculator = new DigestCalculator();
         }
     }
 
+    private void initMaxEphemeralNodesPerSession() {
+        this.ephemeralNodesPerSessionMap = new ConcurrentHashMap<>();
+        maxEphemeralNodesPerSession = Integer.getInteger(ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION, ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION_DEFAULT);
+        if (maxEphemeralNodesPerSession < 0) {
+            maxEphemeralNodesPerSession = ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION_DEFAULT;
+            LOG.warn("maxEphemeralNodesPerSession should be greater than or equal to 0, using default vlaue {}.",
+                ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION_DEFAULT);
+        } else if (maxEphemeralNodesPerSession == ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION_DEFAULT) {
+            LOG.warn("maxEphemeralNodesPerSession is not configured, using default value {}.",
```

Comment:
This may not be necessary IMO, maybe we could just log a INFO level message in Line 141?